tests/http/clients/h2-download.c

```diff
@@ -84,11 +84,12 @@ static int debug_cb(CURL *handle, curl_infotype type,
 
   if(!curl_easy_getinfo(handle, CURLINFO_XFER_ID, &xfer_id) && xfer_id >= 0) {
     if(!curl_easy_getinfo(handle, CURLINFO_CONN_ID, &conn_id) &&
-        conn_id >= 0) {
-      snprintf(idsbuf, sizeof(idsbuf), TRC_IDS_FORMAT_IDS_2, xfer_id, conn_id);
+       conn_id >= 0) {
+      curl_msnprintf(idsbuf, sizeof(idsbuf), TRC_IDS_FORMAT_IDS_2, xfer_id,
+                     conn_id);
     }
     else {
-      snprintf(idsbuf, sizeof(idsbuf), TRC_IDS_FORMAT_IDS_1, xfer_id);
+      curl_msnprintf(idsbuf, sizeof(idsbuf), TRC_IDS_FORMAT_IDS_1, xfer_id);
```

Comment:
Turned these into `snprintfs` in 232302f88a152a1d1722da9f69c383a766528918 [#14382](https://github.com/curl/curl/pull/14382/commits) after a series of whack-a-mole with format failure. Couldn't figure out a better way. (From a build perspective these resemble 'examples', where `snprintf()` is also used as-is.)

Reply:
If we can detect when to.

Reply:
Since this warning only happens with old mingw compilers, it feels like it could just as well be a quirk in those old versions and chasing that down sounds like work with very little benefit. We can just as well disable the printf format attributes for those compilers.

Reply:
... except that I seem to fail at that too.